Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs

```diff
@@ -0,0 +1,346 @@
+            }
+            catch (Exception ex)
+            {
+                GamingCenterType = 0;
+            }
+            return GamingCenterType;
+        }
+
+        
+
+        public static object RegistrySoftwareKeyRead(RegistryHive hKey, string SubKey, string Name, object defaultvalue)
+        {
+            if (Environment.Is64BitOperatingSystem)
+                return RegistryKey.OpenBaseKey(hKey, RegistryView.Registry64).OpenSubKey("SOFTWARE" + SubKey).GetValue(Name, defaultvalue);
+            else
+                return RegistryKey.OpenBaseKey(hKey, RegistryView.Registry32).OpenSubKey("SOFTWARE\\WOW6432Node" + SubKey).GetValue(Name, defaultvalue);
+        }
+      
+
+        public bool CheckGCPower()
+        {
+           if(GamingCenterType==1)
+           {
+                string m_sRegPathI = @"\OEM\Aurora";
+                int Control = (int)RegistrySoftwareKeyRead(RegistryHive.LocalMachine, m_sRegPathI, "AuroraSwitch", 0);
+                if (Control == 0)
+                    return false;
+                else
+                    return true;
+            }
+            
+            else 
+              return true;
+        }
+   
+
+        private void OnRegChanged(object sender, EventArgs e)
+        {
+            bRefreshOnce = true;
+            if (CheckGCPower())
+            {
+                Initialize();
+            } 
+            else
+            {
+                isInitialized = false;
+                Shutdown();
+             
+            }
+            
+        }
+
+        public string GetDeviceName()
+            {
+                return devicename;
+            }
+
+            public string GetDeviceDetails()
+            {
+                if (isInitialized)
+                {
+                    return devicename + ": Initialized";
+                }
+                else
+                {
+                    return devicename + ": Not initialized";
+                }
+            }
+
+            public bool Initialize()
+            {
+           
+
+              if (!isInitialized && CheckGCPower())
+                {
+                    try
+                    {
+                        bRefreshOnce = true;
+                        keyboard =   KeyboardFactory.CreateHIDDevice("hidkeyboard");
+                   
+                    if (keyboard!=null)
+                    {
+                            isInitialized = true;
+                        //SetBrightness();
+                        return true;
+                      }
+                           
+                    
+                    isInitialized = false;
+                    return false;
+                    // Mark Initialized = TRUE
+                   
+                    }
+                    catch (Exception ex)
+                    {
+                        Global.logger.Error("keyboard device, Exception! Message:" + ex);
+                    }
+                    //try
```

Comment:
does this commented code need to be here ?

Reply:
We don't need to use Global.logger. I will remove it.